### PR TITLE
runtime: surpport cold addition vfio for clh

### PR DESCRIPTION
**Pull request by liangxianlong**
In some scenarios, cold addition of vfio network cards is required.

Fixes: #6854

### CODE REVIEW

**Review by likebreath — COMMENTED** [review]
Code changes look good to me. Can you please extend our unit tests (from `clh_test.go`) to exercise this added function? Thanks.

**Comment by liangxianlong** [comment]
ok

**Comment by liangxianlong** [comment]
@likebreath

File: src/runtime/virtcontainers/clh_test.go [hunk 1]
```diff
@@ -127,6 +127,25 @@ func TestCloudHypervisorAddVSock(t *testing.T) {
...
+	testCloudHypervisorAddDevice(t, vfDevice)
+}
+
+func testCloudHypervisorAddDevice(t *testing.T, structure interface{}) {
+	assert := assert.New(t)
+	clh := cloudHypervisor{}
+	switch s := structure.(type) {
+	case config.VFIODev:
+		err := clh.addVfioDevice(s)
+		assert.Nil(err)
```
Comment (likebreath) [inline src/runtime/virtcontainers/clh_test.go:145]: I think the the test will always pass as the new function `addVfioDevice()` will never return `error`. 

A more meaningful test should assert on whether the content of added VFIO device (e.g. `clh.vmconfig.DeviceConfig`) is as expected, say check if its `Path` and `Id` is execpted .

**Comment by amshinde** [comment]
@liangxianlong Can you squash the first two commits and remove the merge commits?

**Pull request closed without merging**